### PR TITLE
VACMS-13226: explicitly set webhost for links in notification email

### LINKED ISSUE #13226: Outdated content notifications contained invalid URL.
## Describe the defect

See [this thread](https://dsva.slack.com/archives/CT4GZBM8F/p1681217460151819).

Outdated content notifications send from the production environment can apparently contain `default` instead of `prod.cms.va.gov`. Unknown why exactly this is.

## Acceptance Criteria

- [ ] make explicitly set the domain to default to prod (instead of Drupal default)
- [ ] make it dynamic to local environment  (for testing purposes)

### PR DESCRIPTION
**Pull request by edmund-dunn**
## Description

Closes #13226.

## Testing done
Tested locally and on tugboat. Queue the notifications and run cron. Then inspect the caught emails for the correct links.

## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

1. Shell into the tugboat and run `drush php-eval "print_r(\Drupal::service('va_gov_notifications.outdated_content')->checkForOutdatedVamcContent());"`
   - [x] Validate that notifications are queued at `/admin/config/system/queues`
2. Navigate to `/admin/config/system/cron` and then go tot he main tugboat admin page
   - [x] Validate that the caught emails contain the correct host in the links tot he outdated content tool and the KB article.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [x] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [x] No announcement is needed for this code change.
  - [x] Merge & carry on unburdened by announcements

### CODE REVIEW

**Review by ndouglas — COMMENTED** [review]
**Question**: Did the environment discovery service not suffice for this use case? I feel this is repeating work we have done elsewhere.

**Comment by edmund-dunn** [comment]
> **Question**: Did the environment discovery service not suffice for this use case? I feel this is repeating work we have done elsewhere.

It didn't, it would tell me the environment but not the host. Since I needed the host and that info was already contained in the settings.php files, I just made use of it.

File: docroot/modules/custom/va_gov_notifications/va_gov_notifications.module [hunk 1]
```diff
@@ -5,8 +5,8 @@
  * Primary module hooks for VA.gov Notifications module.
  */
 
-use Drupal\media\Entity\Media;
-use Drupal\file\Entity\File;
+use Drupal\Core\Site\Settings;
+
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_notifications/va_gov_notifications.module:9]:
```suggestion
```

Comment (ndouglas) [inline docroot/modules/custom/va_gov_notifications/va_gov_notifications.module:9]: pls remove this line

**Review by ndouglas — DISMISSED** [review]
This looks great. I'd love to be able to test it on BRD -- do you know of any way to do that?

**Review by ndouglas — APPROVED** [review]

**Pull request merged**